### PR TITLE
Allocate memory for optional password when needed

### PR DESCRIPTION
**Pull request by bjosv**

### CODE REVIEW

File: hircluster.h [hunk 1]
```diff
@@ -113,11 +111,11 @@ typedef struct redisClusterContext {
...
-    int flags;                                  /* Configuration flags */
-    struct timeval *connect_timeout;            /* TCP connect timeout */
-    struct timeval *command_timeout;            /* Receive and send timeout */
-    int max_retry_count;                        /* Allowed retry attempts */
-    char password[CONFIG_AUTHPASS_MAX_LEN + 1]; /* Include a null terminator */
+    int flags;                       /* Configuration flags */
+    struct timeval *connect_timeout; /* TCP connect timeout */
+    struct timeval *command_timeout; /* Receive and send timeout */
+    int max_retry_count;             /* Allowed retry attempts */
+    char *password;                  /* Authentication password */
```
Comment (zuiderkwast) [inline hircluster.h:118]: Why move the comment alignment column? (Doesn't matter but I just don't see why.)

**Review by zuiderkwast — APPROVED** [review]
LGTM

**Comment by zuiderkwast** [comment]
The commit message looks like it will allocate more memory, i.e. something bad, while the purpose of this is to *not* allocate this memory when it's *not* needed. Maybe you can make the commit message look like something good? Maybe just add two "not"s.

File: hircluster.h (hunk 1 above)
Comment (bjosv) [reply hircluster.h:118]: Its auto-formatted this way (`make format`)

**Pull request merged**